src/test/java/gil/server/HandlerTest.java

```diff
@@ -0,0 +1,15 @@
+package gil.server;
+
+import org.junit.Test;
+import java.io.UnsupportedEncodingException;
+import static org.junit.Assert.*;
+
+public class HandlerTest {
+    Handler handler = new Handler();
+
+    @Test
+    public void shouldReturnAResponseObject() throws UnsupportedEncodingException {
+        String emptyRequest = "GET / HTTP/1.1\n\n";
+        assertTrue (handler.processRequest(emptyRequest) instanceof Response);
+    }
```

Comment:
I think a better test here would be to test that the `Response` is returned with the correct fields. If you wrote two tests (one with an empty request and another with a non-empty request) and then tested to see if they were both just instances of the `Response` class, you wouldn't catch the specific implementation of each `Response` object as it should be returned. This is a testing gap because you don't know exactly what `Response` looks like.

Reply:
Instead of using the assertion that a method returns an `instance of Response`, I'd rather see you create the correct response object, set the expected fields on it, then compare it to the actual returned object from the method. Like this:
```String data = "GET / HTTP/1.1 text/html\r\n";
  StringReader dataStream = new StringReader(data);
  Response expectedResponse = new Response();
  expectedResponse.setBody("someBody");
  expectedResponse.setStatusCode("200");
  
  BufferedReader bufferedReader = new BufferedReader(dataStream);
  Response result = handler.processRequest(bufferedReader(bufferedReader);

  assertEquals("200", result.getStatusCode());
  assertEquals("someBody", result.getBody());
```